app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala

```diff
@@ -31,9 +31,45 @@ import javax.inject.Inject
 
 class MessagesController @Inject() (cc: ControllerComponents)(implicit val materializer: Materializer) extends BackendController(cc) with StreamingParsers {
 
-  def post(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
+  def channelResponseGB(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
     request: Request[Source[ByteString, _]] =>
       request.body.runWith(Sink.ignore)
-      Accepted
+
+      validateHeaders(request.headers, "gb") match {
+        case true => Ok
+        case _    => BadRequest
```

Comment:
Can we please add a message to the bad request to indicate what error is being thrown